src/popup/components/Modals/SpendSuccess.vue

```diff
@@ -88,45 +102,61 @@ export default {
     BtnMain,
   },
   props: {
-    resolve: { type: Function, required: true },
-    transaction: { type: Object, required: true },
+    resolve: { type: Function as PropType<() => void>, required: true },
+    transaction: { type: Object as PropType<ITransaction>, required: true },
   },
-  data: () => ({
-    ExternalLink,
-    hideAvatar: false,
-    nameRecipient: null,
-  }),
-  computed: {
-    ...mapState('fungibleTokens', ['availableTokens']),
-    ...mapGetters(['getTxAmountTotal', 'getTxSymbol', 'getExplorerPath']),
-    ...mapGetters('names', ['getPreferred']),
-    isAe() {
-      return !(this.transaction.tx.contractId
-        && this.availableTokens[this.transaction.tx.contractId]?.symbol);
-    },
-    payload() {
-      return getPayload(this.transaction);
-    },
-    hasPayload() {
-      return this.payload && this.payload.length;
-    },
-  },
-  async mounted() {
-    const { recipientId } = this.transaction.tx;
-    await watchUntilTruthy(() => this.$store.state.middleware);
-    if (recipientId.includes('nm_')) {
-      this.hideAvatar = true;
-      this.nameRecipient = (await this.$store.state.middleware.getNameById(recipientId)).name;
+  setup(props, { root }) {
+    const {
+      getTxAmountTotal,
+      getTxSymbol,
+      loadTokenBalances,
+    } = useFungibleTokens({
+      store: root.$store,
+      accountAddress: props.transaction.tx.callerId,
+    });
+    const middleware = useState('middleware');
+    const getExplorerPath = useGetter('getExplorerPath');
+    const getPreferred = useGetter('names/getPreferred');
+
+    const hideAvatar = ref<boolean>(false);
+    const nameRecipient = ref<string>();
+
+    const isAe = computed((): boolean => (
+      getTxSymbol(props.transaction) === AETERNITY_SYMBOL
+    ));
+
+    const payload = computed(() => getPayload(props.transaction));
+    const hasPayload = computed((): boolean => payload.value?.length);
+
+    function closeModal() {
+      loadTokenBalances();
```

Comment:
Since you yourself mentioned that token balances are loading not so fast, i think we can omit this `loadTokenBalances`. The updated amount will be uploaded in a general poll interval.